p2p/switch.go

```diff
@@ -495,13 +520,46 @@ func (sw *Switch) IsDialingOrExistingAddress(addr *NetAddress) bool {
 		(!sw.config.AllowDuplicateIP && sw.peers.HasIP(addr.IP))
 }
 
+// AddPersistentPeers allows you to set persistent peers. It ignores
+// ErrNetAddressLookup. However, if there are other errors, first encounter is
+// returned.
+func (sw *Switch) AddPersistentPeers(addrs []string) error {
+	sw.Logger.Info("Adding persistent peers", "addrs", addrs)
+	netAddrs, errs := NewNetAddressStrings(addrs)
+	// report all the errors
+	for _, err := range errs {
+		sw.Logger.Error("Error in peer's address", "err", err)
+	}
+	// return first non-ErrNetAddressLookup error
+	for _, err := range errs {
+		if _, ok := err.(ErrNetAddressLookup); ok {
+			continue
+		}
+		return err
+	}
+	sw.persistentPeersAddrs = netAddrs
+	return nil
+}
+
+func (sw *Switch) isPeerPersistentFn() func(*NetAddress) bool {
```

Comment:
I wonder if this should just check the ID instead. Since for instance if there's DNS involved the IP address could change between when it got added to persistent peers and when we connected to it and then this would fail.

Reply:
Also why is this a closure?

Reply:
> I wonder if this should just check the ID instead. Since for instance if there's DNS involved the IP address could change between when it got added to persistent peers and when we connected to it and then this would fail.

This deserves a separate issue. One argument is favor of IP addresses is that it's more static. Maybe there are some DNS attacks we're not considering.

Reply:
> Also why is this a closure?

I thought it would be weird to pass persistent_peers list to `peerConfig`.

Reply:
https://github.com/tendermint/tendermint/issues/3692